### PR TITLE
feat(useful-packages): Replace `moment.js` with `Day.js`.

### LINKED ISSUE #682: remove momentJS from Usefull packages
**Is your feature request related to a problem? Please describe.**
momentJS is no longer under active development. Please remove this recommendation from the docs. They recommend not to use it for new projects anymore.
- [momentJS Project status](https://momentjs.com/docs/#/-project-status/)
- [https://deploy-preview-638--discordjs-guide.netlify.app/miscellaneous/useful-packages.html#moment-js](https://deploy-preview-638--discordjs-guide.netlify.app/miscellaneous/useful-packages.html#moment-js)

**Describe the ideal solution**
Maybe change it to something like Day.js or any other lib.

### PR DESCRIPTION
**Pull request by WiseDevHelper**
**Please describe the changes this PR makes and why it should be merged:**

Resolves #682.

[Demo](https://deploy-preview-685--discordjs-guide.netlify.app/)

This PR replaces the mutable `moment.js` package with the immutable `Day.js` package in the `Useful packages` section in the guide. The goal of this PR is to make sure developers can use the more secure module without the vulnerabilities that `moment.js` has.

**Status**
- [x] Replace `moment.js` with `Day.js`.

### CODE REVIEW

File: guide/miscellaneous/useful-packages.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
 ::: tip
-Official documentation: [https://momentjs.com/docs](https://momentjs.com/docs)
+Official documentation: [https://day.js.org/](https://day.js.org/en)
 :::
 
-Moment is a powerful package for working with dates in JavaScript.  
+Day.js is a powerful package that parses, validates, manipulates, and displays dates and times in JavaScript.
 It allows you to quickly and easily format dates in any way you want or parse strings back into JavaScript Date objects.  
-There are even some extensions for it to allow you to work with durations and more.
+There are even some plugins for it to allow you to work with durations and more.
```
Comment (Ayfri) [inline guide/miscellaneous/useful-packages.md:11]: Day.js without any plugin is poor on features, so you should say that it often requires plugins.

Comment (WaitroseDev) [reply guide/miscellaneous/useful-packages.md:11]: Surely a better option would be to use a lib such as date-fns? Obviously, there are many different date-time libraries, but I've personally found date-fns to be one of the better ones.

Comment (WiseDevHelper) [reply guide/miscellaneous/useful-packages.md:11]: Had talked with @Danktuary regarding this earlier, and we decided to use `Day.js` in the guide.

Comment (Danktuary) [reply guide/miscellaneous/useful-packages.md:11]: For the purposes and examples listed in this page, I think day.js is fine. It has a simple API, is small in size, and has plugins for common use cases.

File: guide/miscellaneous/useful-packages.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
+	max: 1,
 	time: 10e3,
 	errors: ['time'],
 });
-const date = moment(input.first().content);
+const date = dayjs(input.first().content).toDate();
 ```
 
-Using the "moment-duration-format" extension, you could tell the user how many days in the future or past the date is:
+Using the "Duration" plugin, you could tell the user how many days in the future or past the date is:
```
Comment (Danktuary) [inline guide/miscellaneous/useful-packages.md:26]:
```suggestion
Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user how many days in the future or past the date is:
```

File: guide/miscellaneous/useful-packages.md [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
+	max: 1,
 	time: 10e3,
 	errors: ['time'],
 });
-const date = moment(input.first().content);
+const date = dayjs(input.first().content).toDate();
 ```
 
-Using the "moment-duration-format" extension, you could tell the user how many days in the future or past the date is:
+Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user how many days in the future or past the date is:
```
Comment (Danktuary) [inline guide/miscellaneous/useful-packages.md:26]:
```suggestion
Using the [duration plugin](https://day.js.org/docs/en/durations/durations), you could tell the user if the date is in the future or the past:
```
Didn't notice this at first but this sentence doesn't exactly line up with the codeblock below.

**Review by Danktuary — APPROVED** [review]

**Pull request merged**